### PR TITLE
feat: Add batch processing for eachAsync

### LINKED ISSUE #9861: Feature: Add forAll for cursor queries
**Do you want to request a *feature* or report a *bug*?**
Feature

### Feature Description

For cursor queries (that return the data in batches) currently there is a method `forEach` which is used to loop over **each** document returned from the query **not** each batch. it would be nice to loop over **each batch** so you can perform a bulk action on each batch.

### Use Case
You want to send an email to all users currently in your system obviously you will fetch the whole user collection in batches. But it's not very efficient  to send each user an email one at a time that wouldn't be very cost effective plus any mail service has a limit on the max number of bulk emails that could be sent (1000 for example). So the ideal situation would be to fetch the users in batches of 1000 and for  **each batch**  perform a bulk action of sending an email to all those users.

I can make the PR myself but I will wait for the issue approval

### PR DESCRIPTION
**Pull request by khaledosama999**
Resolves #9861

### CODE REVIEW

**Comment by khaledosama999** [comment]
Not the proudest code I've written, but it does the job with minimal changes but honestly, I would prefer for it to be a standalone method.

**Comment by khaledosama999** [comment]
I can rebase the feature on the master branch and change the base of the PR to master if you want

**Comment by khaledosama999** [comment]
@vkarpov15

**Review by miladezzat — APPROVED** [review]

**Review by miladezzat — COMMENTED** [review]
Good Job Mr @khaledosama999 , I think it will be a good feature for me

File: lib/helpers/cursor/eachAsync.js [hunk 1]
```diff
@@ -66,8 +70,25 @@ module.exports = function eachAsync(next, fn, options, callback) {
...
         process.nextTick(() => done());
 
-        handleNextResult(doc, currentDocumentIndex++, function(err) {
-          --handleResultsInProgress;
+        if (batchSize) {
+          documentsBatch.push(doc);
+        }
+
+        // If the current documents size is less than the provided patch size don't process the documents yet
+        if (batchSize && documentsBatch.length !== batchSize) {
```
Comment (vkarpov15) [inline lib/helpers/cursor/eachAsync.js:78]: I'd do `documentsBatch.length < batchSize` just in case, but I don't see that being a blocker here.

**Review by vkarpov15 — APPROVED** [review]
LGTM :+1: I'll merge this into the 5.12 branch and release this with 5.12.0 unless you have any objections?

**Comment by khaledosama999** [comment]
@vkarpov15  No fine by me, like I said if you want me to rebase the feature on another branch that's fine.

**Comment by vkarpov15** [comment]
My mistake, I didn't notice this was against the 6.0 branch. Would it be possible for you to rebase this against the 5.12 branch? 6.0 is a major backwards breaking release that we plan on shipping in early 2021, 5.12 is a feature release we're planning on shipping in Feb '21.

**Comment by khaledosama999** [comment]
No problem 
give me a few mins

**Comment by khaledosama999** [comment]
@vkarpov15  Done, no idea why Travis CI failed though.

**Review by vkarpov15 — APPROVED** [review]
Our travis config is unfortunately long out of date and we haven't taken the time to fix it yet. The rest of the tests passed, so LGTM

**Pull request merged**